open_city_profile/views.py

```diff
@@ -0,0 +1,66 @@
+import sentry_sdk
+from django.core.exceptions import ObjectDoesNotExist, PermissionDenied
+from graphene_django.views import GraphQLView as BaseGraphQLView
+
+from open_city_profile.consts import (
+    GENERAL_ERROR,
+    OBJECT_DOES_NOT_EXIST_ERROR,
+    PERMISSION_DENIED_ERROR,
+    SERVICE_ALREADY_EXISTS_ERROR,
+)
+from open_city_profile.exceptions import ProfileGraphQLError, ServiceAlreadyExistError
+
+error_codes_shared = {
+    Exception: GENERAL_ERROR,
+    PermissionDenied: PERMISSION_DENIED_ERROR,
+    ObjectDoesNotExist: OBJECT_DOES_NOT_EXIST_ERROR,
+}
+error_codes_profile = {ServiceAlreadyExistError: SERVICE_ALREADY_EXISTS_ERROR}
+error_codes = {**error_codes_shared, **error_codes_profile}
+
+
+class GraphQLView(BaseGraphQLView):
+    def execute_graphql_request(self, request, data, query, *args, **kwargs):
+        """Extract any exceptions and send some of them to Sentry"""
+        result = super().execute_graphql_request(request, data, query, *args, **kwargs)
+        # If 'invalid' is set, it's a bad request
+        if result and result.errors and not result.invalid:
+            errors = [
+                e
+                for e in result.errors
+                if not isinstance(
+                    getattr(e, "original_error", None), ProfileGraphQLError
+                )
+            ]
+            if errors:
+                self._capture_sentry_exceptions(result.errors, query)
+        return result
+
+    def _capture_sentry_exceptions(self, errors, query):
+        with sentry_sdk.configure_scope() as scope:
+            scope.set_extra("graphql_query", query)
+            for error in errors:
+                if hasattr(error, "original_error"):
+                    error = error.original_error
+                sentry_sdk.capture_exception(error)
+
+    @staticmethod
+    def format_error(error):
+        def get_error_code(exception):
+            """Get the most specific error code for the exception via superclass"""
+            for exception in exception.mro():
+                try:
+                    return error_codes[exception]
+                except KeyError:
+                    continue
+
+        try:
+            error_code = get_error_code(error.original_error.__class__)
+        except AttributeError:
+            error_code = GENERAL_ERROR
+        formatted_error = super(GraphQLView, GraphQLView).format_error(error)
+        if error_code and (
+            isinstance(formatted_error, dict) and "extensions" not in formatted_error
```

Comment:
Extensions might be there because you can throw a GraphQLError and give it `extensions` parameter which would be set here. I think this is useful in cases where you wan't to add all kind of stuff in the extensions (as in spec: https://helsinkisolutionoffice.atlassian.net/wiki/spaces/DD/pages/59965445/GraphQL+guidelines#GraphQL-error-message-format) and cannot rely solely to the exception/error message mapping.

I still think that the if-clause was lacking checking if the actual `code` exists. That's now fixed.

Reply:
Ok, and now with the code check it makes sense, as it is now possible to add something extra to `extensions` and still get the code there automagically :+1:

I still think it would be nicer if the check was at the start of the method, so it wouldn't figure out the code only to find out later that it is not used for anything, but this is getting nitpicky I guess :)